File: packages/ghidra-mcp/ghidra_scripts/ExportFunctions.py
# Ghidra Python script to export function info
# @category Analysis

from ghidra.program.model.listing import FunctionIterator
from ghidra.program.model.symbol import SymbolType
import os

out_dir = os.path.expanduser("~/hamsterball-re/analysis/ghidra/")

# Export function list
func_path = out_dir + "functions.txt"
str_path = out_dir + "strings.txt"

fm = currentProgram.getFunctionManager()

with open(func_path, "w") as f:
    f.write("# Hamsterball.exe Functions\n")
    f.write("# Address\tName\tSize\tPrototype\n")
    count = 0
    for func in fm.getFunctions(True):
        addr = hex(func.getEntryPoint().offset)
        name = func.getName()
        size = func.getBody().getNumAddresses()
        sig = func.getSignature().getPrototypeString()
        f.write(f"{addr}\t{name}\t{size}\t{sig}\n")
        count += 1
        if count > 5000:
            break
    print(f"Exported {count} functions")

# Export strings
listing = currentProgram.getListing()
count = 0
with open(str_path, "w") as f:
    f.write("# Hamsterball.exe Defined Strings\n")
    f.write("# Address\tValue\n")
    for data in listing.getDefinedData(True):
        if data.hasStringValue():
            val = str(data.getValue())
            if len(val) > 2:
                addr = hex(data.getAddress().offset)
                f.write(f"{addr}\t{val}\n")
                count += 1
                if count > 5000:
                    break
    print(f"Exported {count} strings")

print("Done!")

File: packages/ghidra-mcp/ghidra_scripts/ExportFunctions.java
// Ghidra script to export function list and string references
//@category Analysis
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.*;
import ghidra.program.model.symbol.*;
import ghidra.program.model.address.*;
import java.io.*;

public class ExportFunctions extends GhidraScript {
    @Override
    public void run() throws Exception {
        String outDir = System.getProperty("user.home") + "/hamsterball-re/analysis/ghidra/";
        
        // Export function list
        PrintWriter funcWriter = new PrintWriter(new FileWriter(outDir + "functions.txt"));
        funcWriter.println("# Hamsterball.exe Functions");
        funcWriter.println("# Address\tName\tSize\tPrototype");
        
        FunctionManager fm = currentProgram.getFunctionManager();
        FunctionIterator iter = fm.getFunctions(true);
        int count = 0;
        while (iter.hasNext() && count < 5000) {
            Function f = iter.next();
            funcWriter.printf("0x%08X\t%s\t%d\t%s\n",
                f.getEntryPoint().offset,
                f.getName(),
                f.getBody().getNumAddresses(),
                f.getSignature().getPrototypeString()
            );
            count++;
        }
        funcWriter.close();
        println("Exported " + count + " functions to functions.txt");
        
        // Export string references
        PrintWriter strWriter = new PrintWriter(new FileWriter(outDir + "strings.txt"));
        strWriter.println("# Hamsterball.exe Defined Strings");
        strWriter.println("# Address\tValue");
        
        DataIterator dataIter = currentProgram.getListing().getDefinedData(true);
        count = 0;
        while (dataIter.hasNext() && count < 5000) {
            Data d = dataIter.next();
            if (d.hasStringValue()) {
                String val = d.getValue().toString();
                if (val.length() > 2) {
                    strWriter.printf("0x%08X\t%s\n", d.getAddress().offset, 
                        val.replace("\n", "\\n").replace("\t", "\\t"));
                    count++;
                }
            }
        }
        strWriter.close();
        println("Exported " + count + " strings to strings.txt");
    }
}